pkg/nftables/nftables.go

```diff
@@ -0,0 +1,204 @@
+package nftables
+
+import (
+	"errors"
+	"fmt"
+	"log/slog"
+	"net"
+
+	"github.com/google/nftables"
+	"github.com/google/nftables/expr"
+	"golang.org/x/sys/unix"
+)
+
+var Accept = nftables.ChainPolicyAccept
+
+const (
+	NatTable  = "kube_vip"
+	SNatChain = "kube_vip_snat"
+	DNatChain = "kube_vip_dnat"
+)
+
+func ApplySNAT(podIP, vipIP string, ignoreCIDR []string, IPv6 bool) error {
+
+	conn, err := nftables.New()
+	if err != nil {
+		panic(err)
+	}
+
+	err = Flush(conn, IPv6) // TODO: pass in the type of table we need to concern ourselves with
+	if err != nil {
+		panic(err)
+	}
+
+	rule, err := CreateRule(podIP, vipIP, ignoreCIDR, conn, IPv6)
+	if err != nil {
+		panic(err)
+	}
+	slog.Debug("[egress]", "table", rule.Table.Name, "chain", rule.Chain.Name, "expr", rule.Exprs)
+	conn.AddRule(rule) // Add the rule
+	conn.Flush()       // Commit the rule to nftables
+	err = conn.CloseLasting()
+	if err != nil {
+		panic(err)
+	}
+	return nil
+}
+
+func GetTable(IPv6 bool) *nftables.Table {
+	// Default to IPv4
+	table := &nftables.Table{
+		Family: nftables.TableFamilyIPv4,
+		Name:   NatTable,
+	}
+
+	// Move to IPv6 if needed
+	if IPv6 == true {
+		table.Family = nftables.TableFamilyIPv6
+	}
+	return table
+}
+
+func GetSNatChain(IPv6 bool) *nftables.Chain {
+	policy := nftables.ChainPolicyAccept
+	return &nftables.Chain{
+		Name:     SNatChain,
+		Table:    GetTable(IPv6),
+		Type:     nftables.ChainTypeNAT,
+		Hooknum:  nftables.ChainHookPostrouting,
+		Priority: nftables.ChainPriorityNATSource,
+		Policy:   &policy,
+	}
+}
+
+func FilterTable(conn *nftables.Conn, tableName string) (table *nftables.Table, err error) {
```

Comment:
nit: the return parameters here are inconsistent with the rest of package, those are named values :) 

as you are not even using the "table" name below, maybe remove the named returns here as well?